File: clients/auth-sample/src/lib/services/auth/error-codes.ts
export const AuthErrorCodes = {
    DuplicateEmail: "1001",
    UserNotFound: "1002",
    VerificationCodeMismatch: "2001",
    VerificationCodeExpired: "2002",
    VerificationAttemptsExceeded: "2003",
    VerificationCodeDeliveryFailed: "2004",
    Unexpected: "9999",
} as const

export type KnownAuthErrorCode = (typeof AuthErrorCodes)[keyof typeof AuthErrorCodes]

File: microservices/Auth/src/Auth.Core/Exceptions/ErrorCodes.cs
namespace AuthSample.Auth.Core.Exceptions;

public static class ErrorCodes
{
    public const string DuplicateEmail = "1001";
    public const string UserNotFound = "1002";

    public const string VerificationCodeMismatch = "2001";
    public const string VerificationCodeExpired = "2002";
    public const string VerificationAttemptsExceeded = "2003";
    public const string VerificationCodeDeliveryFailed = "2004";

    public const string Unexpected = "9999";
}


